tools/deviceGen/input/device_limits.json

```diff
@@ -0,0 +1,2347 @@
+            "name": "maxCombinedClipAndCullDistances",
+            "value": 8
+        },
+        {
+            "type": "uint32_t",
+            "name": "discreteQueuePriorities",
+            "value": 2
+        },
+        {
+            "type": "float",
+            "name": "pointSizeRange[2]",
+            "value": "{1.f,64.f}"
+        },
+        {
+            "type": "float",
+            "name": "lineWidthRange[2]",
+            "value": "{1.f,1.f}"
+        },
+        {
+            "type": "float",
+            "name": "pointSizeGranularity",
+            "value": "1",
+            "compareFlipped": true
+        },
+        {
+            "type": "float",
+            "name": "lineWidthGranularity",
+            "value": "1",
+            "compareFlipped": true
+
+        },
+        {
+            "type": "bool",
+            "name": "strictLines",
+            "value": false,
+            "comment": "old intels can't do this"
+        },
+        {
+            "type": "bool",
+            "name": "standardSampleLocations",
+            "value": false,
+            "comment": "Had to roll back from requiring, ROADMAP 2022 but some of our targets missing"
+        },
+        {
+            "type": "uint16_t",
+            "name": "optimalBufferCopyOffsetAlignment",
+            "value": 256,
+            "compareFlipped": true
+
+        },
+        {
+            "type": "uint16_t",
+            "name": "optimalBufferCopyRowPitchAlignment",
+            "value": 128,
+            "compareFlipped": true
+
+        },
+        {
+            "type": "uint16_t",
+            "name": "nonCoherentAtomSize",
+            "value": 256,
+            "compareFlipped": true
+
+        },
+        {
+            "groupComment": [
+                "TODO: later",
+                "VkPhysicalDeviceSparseProperties"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "residencyStandard2DBlockShape",
+            "value": true,
+            "expose": false
+        },
+        {
+            "type": "bool",
+            "name": "residencyStandard2DMultisampleBlockShape",
+            "value": false,
+            "expose": false
+
+        },
+        {
+            "type": "bool",
+            "name": "residencyStandard3DBlockShape",
+            "value": true,
+            "expose": false
+
+        },
+        {
+            "type": "bool",
+            "name": "residencyAlignedMipSize",
+            "value": false,
+            "expose": false
+
+        },
+        {
+            "type": "bool",
+            "name": "residencyNonResidentStrict",
+            "value": true,
+            "expose": false
+
+        }
+    ],
+    "vulkan11core": [
+        {
+            "groupComment": [
+                "Vulkan 1.1 Core"
+            ]
+        },
+        {
+            "type": "uint16_t",
+            "name": "subgroupSize",
+            "value": "4",
+            "compareExpose": false
```

Comment:
instead of having `compareExpose` and `compareFlipped`, just have a `compare` enum with:
- DEFAULT
- REVERSE
- DONT